VideoLocker/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -51,20 +52,21 @@ public boolean isDownloadOverWifiOnly() {
      */
     public File getDownloadFolder() {
         ProfileModel profile = getProfile();
-        
-        File android = new File(Environment.getExternalStorageDirectory(), "Android");
-        File downloadsDir = new File(android, "data");
-        File packDir = new File(downloadsDir, context.getPackageName());
-        File edxDir = new File(packDir, profile.username);
-        edxDir.mkdirs();
+        File videosDir = new File(context.getExternalFilesDir(null).getParent(), AppConstants.Folders.VIDEOS);
```

Comment:
Your solution seems to be in line with option 2, except that instead of falling back to the internal storage, you're instead constructing an invalid path to an external storage which would not at the time be actually mounted.